internal/mode/static/state/graph/route_common.go

```diff
@@ -354,7 +355,11 @@ func tryToAttachRouteToListeners(
 	rk := CreateRouteKey(route.Source)
 
 	bind := func(l *Listener) (allowed, attached bool) {
-		if !routeAllowedByListener(l, route.Source.GetNamespace(), gw.Source.Namespace, namespaces) {
+		if !isRouteNamespaceAllowedByListener(l, route.Source.GetNamespace(), gw.Source.Namespace, namespaces) {
+			return false, false
+		}
+
+		if !isRouteKindAllowedByListener(l, route.Source.GetObjectKind()) {
```

Comment:
Should we have a different condition for when the kind is not supported?

Reply:
No, the condition NotAllowedByListeners is correct here:

```
// This reason is used with the "Accepted" condition when the route has not
// been accepted by a Gateway because the Gateway has no Listener whose
// allowedRoutes criteria permit the route
RouteReasonNotAllowedByListeners RouteConditionReason = "NotAllowedByListeners"
```

Reply:
So, this function only returns information on if the listener can be `attached` or `allowed` as bool. 

We later attach the condition based on the results in L386.